addon/components/select-component.js

```diff
@@ -262,7 +262,8 @@ export default Ember.Component.extend(
         });
       // clear the query string when dropdown is hidden
       } else {
-        return this.set('value', '');
+        this.set('value', '');
+        this.get('parentView').send('valueChanged', '');
```

Comment:
Shouldn't this be `this.sendAction('valueChanged', '')`?

Reply:
Since this is within the DebouncedTextComponent child view, I'm using `this.get('parentView').send()`, to trigger the action from the SelectComponent parent view. I'm not sure this is the right way to do things, but it's currently done this way on 275-276 in `propagateNewText`